zerver/lib/users.py

```diff
@@ -217,7 +217,7 @@ def user_ids_to_users(user_ids: Sequence[int], realm: Realm) -> List[UserProfile
     # users should be included.
 
     def fetch_users_by_id(user_ids: List[int]) -> List[UserProfile]:
-        return list(UserProfile.objects.filter(id__in=user_ids).select_related())
+        return list(UserProfile.objects.filter(id__in=user_ids).select_related("realm"))
```

Comment:
I think the commit message for this might be wrong -- doesn't `select_related()` also select things like `realm__foo_group_id`?

Reply:
OK so I checked the documentation: https://docs.djangoproject.com/en/4.2/ref/models/querysets/#select-related says "This will follow all non-null foreign keys it can find - nullable foreign keys must be specified.".

So I think `realm__foo` foreign keys were included (unintentionally).

Reply:
Oh but we don't have any yet -- OK this is probably fine.